### PR TITLE
Feature/workspace base

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Initial ``conan workspace`` initial proposal to manage local set of editables. Introduced only as a dev/maintainers feature, behind an environment variable.
Docs: https://github.com/conan-io/docs/pull/3930

### CODE REVIEW

**Review by czoido — APPROVED** [review]

File: conan/cli/commands/workspace.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    Display info for current workspace
+    """
+    parser.parse_args(*args)
+    return {"info": conan_api.workspace.info()}
+
+
+@conan_command(group="Consumer")
+def workspace(conan_api, parser, *args):
+    """
+    Manage the remote list and the users authenticated on them.
```
Comment (aander80) [inline conan/cli/commands/workspace.py]: I guess this docstring is a copy-paste error?

Comment (memsharded) [reply conan/cli/commands/workspace.py]: Yes, copy-paste issue, thanks for pointing it out!

File: test/integration/workspace/test_workspace.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            workspace = textwrap.dedent("""\
+               import os
+               name = "myws"
+
+               def editables(*args, **kwargs):
+                   result = {}
+                   for f in os.listdir(workspace_api.folder):
+                       if os.path.isdir(f):
+                           conanfile = workspace_api.load(os.path.join(f, "conanfile.py"))
+                           result[f"{conanfile.name}/{conanfile.version}"] = {"path": f}
```
Comment (AbrilRBS) [inline test/integration/workspace/test_workspace.py]:
```suggestion
                           conanfile_path = os.path.join(f, "conanfile.py")
                           conanfile = workspace_api.load(conanfile_path)
                           result[f"{conanfile.name}/{conanfile.version}"] = {"path": conanfile_path}
```

found while playing around with the feature with @perseoGI , where we used this test as a starting point

Comment (memsharded) [reply test/integration/workspace/test_workspace.py]: Solved

**Review by czoido — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**